iceoryx_examples/iceperf/iceperf_follower.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2019 - 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2020 - 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceperf_follower.hpp"
+#include "iceoryx.hpp"
+#include "iceoryx_c.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "mq.hpp"
+#include "topic_data.hpp"
+#include "uds.hpp"
+
+#include <iostream>
+
+constexpr const char APP_NAME[]{"iceperf-bench-follower"};
+constexpr const char PUBLISHER[]{"Laurel"};
+constexpr const char SUBSCRIBER[]{"Hardy"};
+
+void IcePerfFollower::doTest(IcePerfBase& ipcTechnology) noexcept
+{
+    ipcTechnology.initFollower();
+
+    ipcTechnology.latencyPerfTestFollower();
+
+    ipcTechnology.shutdown();
+}
+
+PerfSettings IcePerfFollower::getSettings(iox::popo::Subscriber<PerfSettings>& subscriber) noexcept
+{
+    // wait for settings from leader application
+    constexpr bool WAIT_FOR_SETTINGS{true};
+    while (WAIT_FOR_SETTINGS)
+    {
+        static bool waitMessagePrinted = false;
+        if (!waitMessagePrinted)
+        {
+            std::cout << "Waiting for PerfSettings from leader application!" << std::endl;
+            waitMessagePrinted = true;
+        }
+
+        auto sample = subscriber.take();
```

Comment:
Why write 8 lines of unclean code when you could write 3 or 4 lines of clean code with `and_then` / `or_else`?

Reply:
because of the return ;)